src/compute/src/render/join/delta_join.rs

```diff
@@ -461,7 +461,9 @@ where
         let (oks, errs2) = dogsdogsdogs::operators::half_join::half_join_internal_unsafe(
             &updates,
             trace,
-            |time| time.step_back(),
+            |time, antichain| {
+                antichain.insert(time.step_back());
```

Comment:
This doesn't yet give us "precise" compaction as described by the description of [this PR](https://github.com/TimelyDataflow/differential-dataflow/pull/457) yet, right? Do we plan to do this in MZ?

Reply:
It does, it turns out. MZ's use is limited to `Product<u64, PointStamp<u64>>`, or whatever, and that happens to be a [distributive lattice](https://en.wikipedia.org/wiki/Distributive_lattice) for which compaction by the meet of the frontier equals compaction by the frontier.

Reply:
Very interesting, thanks for the explanation!